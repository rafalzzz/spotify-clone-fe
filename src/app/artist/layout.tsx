import { FC, PropsWithChildren } from 'react';

const Artist: FC<PropsWithChildren> = ({ children }) => <>{children}</>;

export default Artist;
