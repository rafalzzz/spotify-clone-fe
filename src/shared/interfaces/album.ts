export interface Album {
  artistId: number;
  artistName: string;
  artworkUrl60: string;
  artworkUrl100: string;
  collectionCensoredName: string;
  collectionExplicitness: string;
  collectionId: number;
  collectionName: string;
  collectionPrice: number;
  collectionType: string;
  collectionViewUrl: string;
  copyright: string;
  country: string;
  currency: string;
  primaryGenreName: string;
  releaseDate: string;
  trackCount: number;
  wrapperType: string;
}
