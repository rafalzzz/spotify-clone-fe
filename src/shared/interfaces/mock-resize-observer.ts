export interface MockResizeObserver {
  observe: jest.Mock;
  unobserve: jest.Mock;
}
