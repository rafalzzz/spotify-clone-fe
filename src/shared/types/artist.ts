export const enum EArtistKeys {
  AMG_ARTIST_ID = 'amgArtistId',
  ARTIST_ID = 'artistId',
  ARTIST_LINK_URL = 'artistLinkUrl',
  ARTIST_NAME = 'artistName',
  ARTIST_TYPE = 'artistType',
  PRIMARY_GENRE_ID = 'primaryGenreId',
  PRIMARY_GENRE_NAME = 'primaryGenreName',
  WRAPPER_TYPE = 'wrapperType',
}

export type Artist = {
  [EArtistKeys.AMG_ARTIST_ID]: number;
  [EArtistKeys.ARTIST_ID]: number;
  [EArtistKeys.ARTIST_LINK_URL]: string;
  [EArtistKeys.ARTIST_NAME]: string;
  [EArtistKeys.ARTIST_TYPE]: string;
  [EArtistKeys.PRIMARY_GENRE_ID]: number;
  [EArtistKeys.PRIMARY_GENRE_NAME]: string;
  [EArtistKeys.WRAPPER_TYPE]: string;
};
