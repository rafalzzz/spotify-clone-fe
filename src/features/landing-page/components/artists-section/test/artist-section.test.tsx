import { render } from '@testing-library/react';

import { Artist } from '@/shared/interfaces/artist';

import { ArtistsSection } from '..';

const mockArtists = [
  {
    artistId: 1,
    artistName: 'Test_Artist1',
  },
  {
    artistId: 2,
    artistName: 'Test_Artist2',
  },
  {
    artistId: 3,
    artistName: 'Test_Artist3',
  },
  {
    artistId: 4,
    artistName: 'Test_Artist4',
  },
  {
    artistId: 5,
    artistName: 'Test_Artist5',
  },
  {
    artistId: 6,
    artistName: 'Test_Artist6',
  },
];

const renderArtistsSection = () => render(<ArtistsSection artists={mockArtists as Artist[]} />);

describe('ArtistsSection', () => {
  it('render component without error', () => {
    const screen = renderArtistsSection();
    expect(screen).toMatchSnapshot();
  });
});
