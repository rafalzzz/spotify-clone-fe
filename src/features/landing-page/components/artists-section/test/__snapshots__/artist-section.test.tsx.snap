// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArtistsSection render component without error 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <section
        class="custom-section"
      >
        <div
          class="custom-section__header"
        >
          <header>
            <a
              class="custom-section__header__title"
              data-testid="custom-section-header"
              href="/most-famous-djs"
            >
              <h2>
                Most Popular DJs
              </h2>
            </a>
          </header>
          <a
            class="custom-section__header__redirection"
            data-testid="custom-section-show-all-items"
            href="/most-famous-djs"
          >
            Show all
          </a>
        </div>
        <ul
          class="custom-section__items"
        >
          <li>
            <a
              class="custom-artist-item"
              data-testid="custom-artist-item-redirection"
              href="/artist/Test_Artist1"
            >
              <div
                class="custom-artist-item__wrapper"
              >
                <div>
                  Test_Artist1
                </div>
              </div>
            </a>
          </li>
          <li>
            <a
              class="custom-artist-item"
              data-testid="custom-artist-item-redirection"
              href="/artist/Test_Artist2"
            >
              <div
                class="custom-artist-item__wrapper"
              >
                <div>
                  Test_Artist2
                </div>
              </div>
            </a>
          </li>
          <li>
            <a
              class="custom-artist-item"
              data-testid="custom-artist-item-redirection"
              href="/artist/Test_Artist3"
            >
              <div
                class="custom-artist-item__wrapper"
              >
                <div>
                  Test_Artist3
                </div>
              </div>
            </a>
          </li>
          <li>
            <a
              class="custom-artist-item"
              data-testid="custom-artist-item-redirection"
              href="/artist/Test_Artist4"
            >
              <div
                class="custom-artist-item__wrapper"
              >
                <div>
                  Test_Artist4
                </div>
              </div>
            </a>
          </li>
          <li>
            <a
              class="custom-artist-item"
              data-testid="custom-artist-item-redirection"
              href="/artist/Test_Artist5"
            >
              <div
                class="custom-artist-item__wrapper"
              >
                <div>
                  Test_Artist5
                </div>
              </div>
            </a>
          </li>
          <li>
            <a
              class="custom-artist-item"
              data-testid="custom-artist-item-redirection"
              href="/artist/Test_Artist6"
            >
              <div
                class="custom-artist-item__wrapper"
              >
                <div>
                  Test_Artist6
                </div>
              </div>
            </a>
          </li>
        </ul>
      </section>
    </div>
  </body>,
  "container": <div>
    <section
      class="custom-section"
    >
      <div
        class="custom-section__header"
      >
        <header>
          <a
            class="custom-section__header__title"
            data-testid="custom-section-header"
            href="/most-famous-djs"
          >
            <h2>
              Most Popular DJs
            </h2>
          </a>
        </header>
        <a
          class="custom-section__header__redirection"
          data-testid="custom-section-show-all-items"
          href="/most-famous-djs"
        >
          Show all
        </a>
      </div>
      <ul
        class="custom-section__items"
      >
        <li>
          <a
            class="custom-artist-item"
            data-testid="custom-artist-item-redirection"
            href="/artist/Test_Artist1"
          >
            <div
              class="custom-artist-item__wrapper"
            >
              <div>
                Test_Artist1
              </div>
            </div>
          </a>
        </li>
        <li>
          <a
            class="custom-artist-item"
            data-testid="custom-artist-item-redirection"
            href="/artist/Test_Artist2"
          >
            <div
              class="custom-artist-item__wrapper"
            >
              <div>
                Test_Artist2
              </div>
            </div>
          </a>
        </li>
        <li>
          <a
            class="custom-artist-item"
            data-testid="custom-artist-item-redirection"
            href="/artist/Test_Artist3"
          >
            <div
              class="custom-artist-item__wrapper"
            >
              <div>
                Test_Artist3
              </div>
            </div>
          </a>
        </li>
        <li>
          <a
            class="custom-artist-item"
            data-testid="custom-artist-item-redirection"
            href="/artist/Test_Artist4"
          >
            <div
              class="custom-artist-item__wrapper"
            >
              <div>
                Test_Artist4
              </div>
            </div>
          </a>
        </li>
        <li>
          <a
            class="custom-artist-item"
            data-testid="custom-artist-item-redirection"
            href="/artist/Test_Artist5"
          >
            <div
              class="custom-artist-item__wrapper"
            >
              <div>
                Test_Artist5
              </div>
            </div>
          </a>
        </li>
        <li>
          <a
            class="custom-artist-item"
            data-testid="custom-artist-item-redirection"
            href="/artist/Test_Artist6"
          >
            <div
              class="custom-artist-item__wrapper"
            >
              <div>
                Test_Artist6
              </div>
            </div>
          </a>
        </li>
      </ul>
    </section>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
